### PR TITLE
Legacy Media Placeholder Block

### LINKED ISSUE #5517: Media placeholder for TC2/legacy content block
We would like to show a placeholder which states that the media is unavailable for legacy content as an interim solution while we build the support for playback in an upcoming release. 

Ideally using something which is currently available, ie. with translations.

See https://paper.dropbox.com/doc/Legacy-Maps-Example-Mapping--AuP~mLw_AtiCcNot_vwN68OfAg-xyqczFWE1Uxp08oH3T3Pq#:uid=766751506940731624298759&h2=Render-Legacy-MAP-page-with-me for further information

- [ ] Include UX in 3-Amigos session

### PR DESCRIPTION
**Pull request by ryanmccombe**
Resolves https://github.com/bbc/simorgh/issues/5517

**Overall change:** 
Add placeholder fallback for legacyMedia block within MAPs

**Code changes:**
- New fixture that includes modern media block in a legacy asset
- Cleanup and extract `MediaMessage` component
- Use MediaMessage for any legacy media block, or for any media block on a legacy asset
- Add Preprocessor for adding `CounterName` to legacy MAP ATI Analytics (was preventing unit testing of the MAP page due to Prop Type validation

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**
- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/index.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ const typesToConvert = {
   list,
   media,
   version,
+  legacyMedia: block => ({ model: {}, ...block }),
```
Comment (ryanmccombe) [inline src/app/lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/index.js:21]:
```suggestion
  legacyMedia: identity,
```

File: src/app/pages/MediaAssetPage/index.jsx [hunk 2]
```diff
@@ -30,14 +33,19 @@ import {
...
+import { RequestContext } from '#contexts/RequestContext';
+
 // Page Handlers
 import withContexts from '#containers/PageHandlers/withContexts';
 import withPageWrapper from '#containers/PageHandlers/withPageWrapper';
 import withError from '#containers/PageHandlers/withError';
 import withLoading from '#containers/PageHandlers/withLoading';
 import withData from '#containers/PageHandlers/withData';
 
+const isLegacyAsset = url => url.split('/').length > 7;
```
Comment (RichardPK) [inline src/app/pages/MediaAssetPage/index.jsx]: Would this logic be useful for other pods? Could be worth extracting into its own file?

Comment (RichardPK) [inline src/app/pages/MediaAssetPage/index.jsx]: Would this logic be useful for other pods? Could be worth extracting into its own file? 

Will this logic capture 100% of legacy assets?

If it isn't appropriate to move this into its own file maybe rename along the lines of 'isLegacyMAP'?

**Review by RichardPK — COMMENTED** [review]
Looks good - see comment below with a couple qs.

**Review by karinathomasbbc — APPROVED** [review]

File: src/app/pages/MediaAssetPage/index.jsx (hunk 2 above)
Comment (jroebu14) [inline src/app/pages/MediaAssetPage/index.jsx]: is this really the best way to tell if is a legacy asset? if it is then carry on!

**Review by jroebu14 — APPROVED** [review]

File: src/app/pages/MediaAssetPage/index.jsx (hunk 2 above)
Comment (rebeccamcginn) [reply src/app/pages/MediaAssetPage/index.jsx]: It was a choice between the number of slashes and checking whether it contained a date:

uzbek.multimedia.2013.12.131221_sectarian_war_bowen.page

pidgin.media_asset.51536047.page

Counting the slashes was simpler but I would be eager to hear other suggestions

Comment (ryanmccombe) [reply src/app/pages/MediaAssetPage/index.jsx]: Feels wrong but it does seem to be the best way - Jane checked with Dan and inferring it from the URL is what he suggested as well

Comment (ryanmccombe) [reply src/app/pages/MediaAssetPage/index.jsx]: 👍 renamed it

we have no natural place to move it - probably not worth it as it's going to be deleted once we get the legacy media working

**Pull request merged**